packages/amplication-client/src/Application/appSettings/DirectoriesSettingsForm.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Snackbar,
+  Panel,
+  EnumPanelStyle,
+  TextField,
+} from "@amplication/design-system";
+import { gql, useMutation, useQuery } from "@apollo/client";
+import { Form, Formik } from "formik";
+import React, { useCallback, useContext } from "react";
+import * as models from "../../models";
+import { useTracking } from "../../util/analytics";
+import { formatError } from "../../util/error";
+import FormikAutoSave from "../../util/formikAutoSave";
+import { validate } from "../../util/formikValidateJsonSchema";
+import { match } from "react-router-dom";
+import PendingChangesContext from "../../VersionControl/PendingChangesContext";
+import "./GenerationSettingsForm.scss";
+import cssNamingConverter from "../../util/cssNamingConverter";
+import FORM_SCHEMA from "./formSchema";
+import { GET_APP_SETTINGS, UPDATE_APP_SETTINGS } from "./GenerationSettingsForm";
+
+type Props = {
+  match: match<{ application: string }>;
+};
+
+type TData = {
+  updateAppSettings: models.AppSettings;
+};
+
+const CLASS_NAME = cssNamingConverter(GenerationSettingsForm.name);
+
+function GenerationSettingsForm({ match }: Props) {
+  const applicationId = match.params.application;
+
+  const { data, error } = useQuery<{
+    appSettings: models.AppSettings;
+  }>(GET_APP_SETTINGS, {
+    variables: {
+      id: applicationId,
+    },
+  });
+
+  const pendingChangesContext = useContext(PendingChangesContext);
+
+  const { trackEvent } = useTracking();
+
+  const [updateAppSettings, { error: updateError }] = useMutation<TData>(
+    UPDATE_APP_SETTINGS,
+    {
+      onCompleted: (data) => {
+        pendingChangesContext.addBlock(data.updateAppSettings.id);
+      }
+    }
+  );
+
+    
+
+  const handleSubmit = useCallback(
+    (data: models.AppSettings) => {
+      const {
+        dbHost,
+        dbName,
+        dbPassword,
+        dbPort,
+        dbUser,
+        authProvider,
+        adminUISettings: { generateAdminUI, adminUIPath },
+        serverSettings: { generateRestApi, generateGraphQL, serverPath },
+      } = data;
+
+      trackEvent({
+        eventName: "updateAppSettings",
+      });
+      updateAppSettings({
+        variables: {
+          data: {
+            dbHost,
+            dbName,
+            dbPassword,
+            dbPort,
+            dbUser,
+            authProvider,
+            adminUISettings: {
+              generateAdminUI,
+              adminUIPath: adminUIPath || ""
+            },
+            serverSettings: {
+              generateRestApi,
+              generateGraphQL,
+              serverPath: serverPath || ""
+            },
+          },
```

Comment:
I think we can use spread on ...data in a way that is easier to maintain, especially since we know that we are only updating some of the properties. 
Also, I think it is worth checking the server-side, and maybe keeping any existing values on the app settings object unchanged if the input is undefined.